### DISCUSSION TITLE
[py] [force_reannounce] how to pass flags from python?

### DISCUSSION THREAD

**Discussion by laur89** [arvidn/libtorrent#7334] [Q&A] [answered]
With [force_reannounce](https://libtorrent.org/reference-Torrent_Handle.html#force-lsd-announce-force-dht-announce-force-reannounce), how would one pass a `ignore_min_interval` flag via `reannounce_flags_t` param in python bindings in order _not_ to honor min-interval, and announce tracker immediately?
Would `torrenthandle.force_reannounce(0, -1, 1)` -- ie passing `int(1)` -- do the trick?

For completeness, here's the libtorrent method signature:
```c++
void force_reannounce(int seconds = 0, int idx = -1, reannounce_flags_t = {}) const;
```

**Comment by arvidn** [comment, marked as answer]
you can see the binding definition here: https://github.com/arvidn/libtorrent/blob/RC_2_0/bindings/python/src/torrent_handle.cpp#L536
the parameters are named, so I believe you should be able to get away with just specifying `flags=`

The flag value is defined here: https://github.com/arvidn/libtorrent/blob/RC_2_0/bindings/python/src/torrent_handle.cpp#L651

and should be accessible to python as `lt.reannounce_flags_t.ignore_min_interval`

**Reply by laur89** [reply]
Ah those are defaults in python as well, so
```python
torrenthandle.force_reannounce(flags=lt.reannounce_flags_t.ignore_min_interval)
```

should be it. Thanks.

Out of curiosity, how would one provide multiple flags, given reannounce() accepted them? Would it be still single parameter passed via `flags`, but value gets merged together via bitwise operation?

**Reply by arvidn** [reply]
you combine flags with bitwise OR. in python (and most languages) this is done with the `|`-operator

**Comment by laur89** [comment]
Bit unrelated, but wouldn't the correct signature in [the docs](https://libtorrent.org/reference-Torrent_Handle.html#force-lsd-announce-force-dht-announce-force-reannounce) be
```c++
void force_reannounce(int seconds = 0, int idx = -1, reannounce_flags_t flags = {}) const;
```
vs current

```c++
void force_reannounce(int seconds = 0, int idx = -1, reannounce_flags_t = {}) const;
```
(note the missing `flags` in current version)

**Reply by arvidn** [reply]
in C++, those signatures are identical. the parameter name is not a necessary part of a signature. Even the argument name is not necessary in the definition, if it's unused.

**Reply by laur89** [reply]
It's not _technically_ wrong, but it's not consistent with some other signatures in the docs is all. Not a major issue, just wanted to point out in case it's something that got missed.
